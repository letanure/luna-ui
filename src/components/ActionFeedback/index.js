import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import hex2Rgba from '../../ultils/hex2Rgba';

/**
 * ActionFeedback component.
 * E.g.:
 * ```html
 * <ActionFeddback>Theta ui lib =D</ActionFeddback>
 * ```
 */
class ActionFeedbackComponent extends React.PureComponent {
  render() {
    const { className } = this.props;

    const ActionFeedback = styled.div`
      width: 100%;
      display: flex;
      border-radius: 2px;
      padding: 20px 35px;
      align-items: center;
      transition: opacity linear .25s;
      color: ${(props) => props.theme.actionFeedbackTextColor};
      ${(props) =>
    `background-color: ${hex2Rgba(`${props.theme.actionFeedbackBackgroundColor}`, 0.9)};`
};
    `;

    return <ActionFeedback className={className}>{this.props.children}</ActionFeedback>;
  }
}

ActionFeedbackComponent.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

ActionFeedbackComponent.defaultProps = {
  children: null,
  className: ''
};

export default ActionFeedbackComponent;
