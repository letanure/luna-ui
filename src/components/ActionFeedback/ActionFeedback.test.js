import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ActionFeedback from './index';

describe('<ActionFeedback />', () => {
  const component = <ActionFeedback>Lebron James has scored!</ActionFeedback>;
  test('ActionFeedback', () => {
    const feedback = shallow(component);

    expect(feedback.render().text()).toEqual('Lebron James has scored!');
  });

  test('toMatchSnapshot', () => {
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
