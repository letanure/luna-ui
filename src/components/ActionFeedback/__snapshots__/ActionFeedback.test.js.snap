// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionFeedback /> toMatchSnapshot 1`] = `
<div
  className="sc-bwzfXH hyRwHX"
>
  Lebron James has scored!
</div>
`;
